src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.Icu.cs

```diff
@@ -143,7 +247,7 @@ private static void ValidateArguments(string strInput, NormalizationForm normali
         /// We walk the string ourselves looking for these bad sequences so we can continue to throw
         /// ArgumentException in these cases.
         /// </summary>
-        private static bool HasInvalidUnicodeSequence(string s)
+        private static bool HasInvalidUnicodeSequence(ReadOnlySpan<char> s)
```

Comment:
Separate from this PR, this method looks ripe for optimizing with SearchValues, skipping all characters < '\ud800' quickly.

Reply:
A few other places with a similar idea (`SearchValues` one does have a fast path):

https://github.com/dotnet/runtime/blob/8fca0a156dcbb1eba5c79e387a6d8f924bfdd3a6/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs#L482

https://github.com/dotnet/runtime/blob/8fca0a156dcbb1eba5c79e387a6d8f924bfdd3a6/src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs#L1394

https://github.com/dotnet/runtime/blob/8fca0a156dcbb1eba5c79e387a6d8f924bfdd3a6/src/libraries/System.Web.HttpUtility/src/System/Web/Util/Utf16StringValidator.cs#L10

Reply:
We can address the optimization on this method in another PR as this already existed code and I am just using it. @MihaZupan could you include optimizing this when optimizing the other methods?